### PR TITLE
Fix mmCIF parsing bug

### PR DESCRIPTION
**Pull request by jgreener64**
On testing the mmCIF parser on the whole PDB some files threw parsing errors. This was due to lines such as:

```
loop_
_pdbx_audit_revision_item.ordinal
_pdbx_audit_revision_item.revision_ordinal
_pdbx_audit_revision_item.data_content_type
_pdbx_audit_revision_item.item
1  5 'Structure model' '_atom_site.B_iso_or_equiv'
2  5 'Structure model' '_atom_site.Cartn_x'
3  5 'Structure model' '_atom_site.Cartn_y'
4  5 'Structure model' '_atom_site.Cartn_z'
```

Here `_atom_site.B_iso_or_equiv` is read as a new key and it messes up the lists. This PR allows entries not in the first column to start with underscores and be read as values not keys. This fixes parsing on 54 PDB entries.

### CODE REVIEW

**Comment by peterjc** [comment]
@lennax @etal @JoaoRodrigues - could one of you review this please?

**Comment by peterjc** [comment]
@jgreener64 can you add a test case for this please?

**Comment by peterjc** [comment]
@jgreener64 I'm happy, thanks - just waiting for one of our structural people to comment.

File: Bio/PDB/MMCIF2Dict.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def __init__(self, filename):
                     n = 0
                     continue
                 elif loop_flag:
-                    if token.startswith("_"):
+                    if token.startswith("_") and (n == 0 or i % n == 0):
```
Comment (lennax) [inline Bio/PDB/MMCIF2Dict.py]: Please add a brief comment on the logic

**Comment by peterjc** [comment]
Is this OK now @lennax? Thanks!

**Comment by peterjc** [comment]
Thank you both!

**Pull request merged**